lib/rails/pagination.rb

```diff
@@ -47,12 +43,20 @@ def _paginate_collection(collection, options={})
       page_header     = ApiPagination.config.page_header
       include_total   = ApiPagination.config.include_total
 
-      headers['Link']          = links.join(', ') unless links.empty?
-      headers[total_header]    = ApiPagination.total_from(collection) if include_total
+      headers['Link'] = links.join(', ') unless links.empty?
       headers[per_page_header] = options[:per_page].to_s
-      headers[page_header]     = options[:page].to_s unless page_header.nil?
+      headers[page_header] = options[:page].to_s unless page_header.nil?
+      headers[total_header] = total_count(collection, options) if include_total
 
       return collection
     end
+
+    def total_count(collection, options)
+      total_count = if ApiPagination.config.paginator == :kaminari
+        paginate_array_options = options[:paginate_array_options]
+        paginate_array_options[:total_count] if paginate_array_options
+      end
+      total_count || ApiPagination.total_from(collection)
+    end
```

Comment:
Instead of doing a variable assignment, could we just return out of an if-else?

```ruby
if ApiPagination.config.paginator == :kaminari
  paginate_array_options = options[:paginate_array_options]
  paginate_array_options[:total_count] if paginate_array_options
else
  ApiPagination.total_from(collection)
end
```